Guard references to sleep states on CC13x2/CC26x2

vanti opened the pull request:
Use of macros such as `SYS_POWER_STATE_SLEEP_2` needs to be guarded by
making sure `CONFIG_SYS_POWER_SLEEP_STATES` is defined. This is causing Shippable to fail on some tests that set `CONFIG_SYS_POWER_MANAGEMENT=y` and `CONFIG_SYS_POWER_SLEEP_STATES=n`.

Full error log: https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/67344/4/console

vanti committed:
drivers: cc13xx_cc26xx: guard references to sleep states

Use of macros such as SYS_POWER_STATE_SLEEP_2 needs to be guarded by
making sure CONFIG_SYS_POWER_SLEEP_STATES is defined.

Signed-off-by: Vincent Wan <vincent.wan@linaro.org>

vanti committed:
power: policy: cc13x2_cc26x2: guard references to sleep states

Use of macros such as SYS_POWER_STATE_SLEEP_2 needs to be guarded by
making sure CONFIG_SYS_POWER_SLEEP_STATES is defined.

Signed-off-by: Vincent Wan <vincent.wan@linaro.org>

pabigot commented on drivers/i2c/i2c_cc13xx_cc26xx.c:
```diff
@@ -206,7 +206,8 @@ static int i2c_cc13xx_cc26xx_transfer(struct device *dev, struct i2c_msg *msgs,
 
-#ifdef CONFIG_SYS_POWER_MANAGEMENT
+#if defined(CONFIG_SYS_POWER_MANAGEMENT) && \
+	defined(CONFIG_SYS_POWER_SLEEP_STATES)
```
Should this be:

    defined(CONFIG_HAS_SYS_POWER_STATE_SLEEP_2)

Because at the moment each sleep level is independently controlled (!!).  Happens the board currently selects that, but if somebody changes it this would break again.

vanti replied:
Thanks. I think the condition should be removed from `subsys/power/policy/policy_residency_cc13x2_cc26x2.c`. In the [review](https://github.com/zephyrproject-rtos/zephyr/pull/21465#pullrequestreview-349642793) of #21465, we made the decision that it'd be ok to assume `CONFIG_HAS_SYS_POWER_STATE_SLEEP_1` and `CONFIG_HAS_SYS_POWER_STATE_SLEEP_2` are set since they are set at SoC level for this platform. It'd be good to stay consistent.

Meanwhile, `defined(CONFIG_HAS_SYS_POWER_STATE_SLEEP_2)` would not fix the problem for the case when `CONFIG_HAS_SYS_POWER_STATE_SLEEP_2` is defined but `CONFIG_SYS_POWER_SLEEP_STATES` isn't. The build error would still occur.

pabigot replied:
>  would not fix the problem for the case when `CONFIG_HAS_SYS_POWER_STATE_SLEEP_2` is defined but `CONFIG_SYS_POWER_SLEEP_STATES` isn't

I would hope that combination would be rejected by Kconfig, but don't think it's worth checking.

pabigot reviewed (commented):
Questioning whether the test should be specified to `CONFIG_HAS_SYS_POWER_STATE_SLEEP_2`.  That condition is currently used in subsys/power/policy/policy_residency_cc13x2_cc26x2.c.

vanti committed:
power: policy: cc13x2_cc26x2: remove unnecessary sleep state conditions

Minor clean up to remove unnecessary references in the code to
CONFIG_HAS_SYS_POWER_STATE_SLEEP_1 and
CONFIG_HAS_SYS_POWER_STATE_SLEEP_2, as they are always defined for this
SoC.

Signed-off-by: Vincent Wan <vincent.wan@linaro.org>

vanti commented:
Removed unnecessary references to `CONFIG_HAS_SYS_POWER_STATE_SLEEP_1 `and `CONFIG_HAS_SYS_POWER_STATE_SLEEP_2` identified in review from @pabigot.
